domain/application/state/unit.go

```diff
@@ -1216,6 +1216,51 @@ func (st *State) GetUnitNamesForApplication(ctx context.Context, uuid coreapplic
 	}), nil
 }
 
+// GetCloudContainerAddresses returns the addresses of the cloud container for the
+// specified unit.
+//
+// The following errors may be returned:
+// - [uniterrors.UnitNotFound] if the unit does not exist
+func (st *State) GetCloudContainerAddresses(ctx context.Context, name coreunit.Name) (network.SpaceAddresses, error) {
+	db, err := st.DB()
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+
+	var addresses []spaceAddress
+	ident := unitName{Name: name}
+	queryCloudContainerAddressesStmt, err := st.Prepare(`
+SELECT    
+    ip.address_value AS &spaceAddress.address_value,
```

Comment:
Same optimisation can be made here as above.

Reply:
Done.